fcrepo-http-api/src/main/java/org/fcrepo/http/api/ContentExposingResource.java

```diff
@@ -539,10 +569,28 @@ private static void evaluateRequestPreconditions(final Request request,
             return;
         }
 
-        final EntityTag etag = new EntityTag(resource.getEtagValue());
-        final Date date = resource.getLastModifiedDate();
+        final EntityTag etag;
+        final Date date;
         final Date roundedDate = new Date();
 
+        // See the related note about the next block of code in the
+        // ContentExposingResource::addCacheControlHeaders function
+        if (resource instanceof NonRdfSourceDescription) {
```

Comment:
Note to self: Java has no real pattern matching, so we might want to install `isRdf`/`isNotRdf` methods on `FedoraResource`.